specifyweb/frontend/js_src/lib/components/AttachmentsBulkImport/ViewAttachmentFiles.tsx

```diff
@@ -0,0 +1,214 @@
+              handleDisambiguate === undefined ? '' : `hover:bg-brand-200`
+            }
+            onClick={handleDisambiguate}
+          >
+            {resolvedRecord?.type === 'matched' ? (
+              <Link.NewTab
+                href={getResourceViewUrl(baseTableName!, resolvedRecord.id)}
+              >
+                {file.parsedName?.toString()}
+              </Link.NewTab>
+            ) : (
+              resolvedRecord !== undefined && (
+                <div>
+                  {keyLocalizationMapAttachment[resolvedRecord.reason] ?? ''}
+                </div>
+              )
+            )}
+          </div>,
+        ],
+
+        attachmentId,
+        canDisambiguate: typeof handleDisambiguate === 'function',
+        isNativeError: resolvedRecord?.type === 'invalid',
+        isRuntimeError,
+      } as const;
+    }
+  );
+
+export function ViewAttachmentFiles({
+  uploadableFiles,
+  baseTableName,
+  uploadSpec,
+  onDisambiguation: handleDisambiguation,
+  onFilesDropped: handleFilesDropped,
+}: {
+  readonly uploadableFiles: RA<PartialUploadableFileSpec>;
+  readonly baseTableName: keyof Tables | undefined;
+  readonly uploadSpec: PartialAttachmentUploadSpec;
+  readonly onDisambiguation:
+    | ((
+        disambiguatedId: number,
+        indexToDisambiguate: number,
+        multiple: boolean
+      ) => void)
+    | undefined;
+  readonly onFilesDropped?: (file: FileList) => void;
+}): JSX.Element {
+  const [disambiguationIndex, setDisambiguationIndex] = React.useState<
+    number | undefined
+  >(undefined);
+
+  const data = React.useMemo(
+    () =>
+      resolveAttachmentDatasetData(
+        uploadableFiles,
+        setDisambiguationIndex,
+        baseTableName
+      ),
+    [uploadableFiles, setDisambiguationIndex, baseTableName]
+  );
+  const headers = React.useMemo(
+    () => ({
+      selectedFileName: commonText.selectedFileName(),
+      fileSize: attachmentsText.fileSize(),
+      record: (
+        <div className="flex min-w-fit items-center gap-2">
+          {baseTableName === undefined ? (
+            formsText.record()
+          ) : (
+            <>
+              <TableIcon label name={baseTableName} />
+              {uploadSpec.staticPathKey === undefined
+                ? ''
+                : defined(
+                    strictGetModel(baseTableName).getField(
+                      staticAttachmentImportPaths[uploadSpec.staticPathKey].path
+                    )
+                  ).label}
+            </>
+          )}
+        </div>
+      ),
+      status: attachmentsText.status(),
+      attachmentId: attachmentsText.attachmentID(),
+    }),
+    [uploadSpec.staticPathKey]
+  );
+
+  const fileDropDivRef = React.useRef<HTMLDivElement>(null);
+  const { isDragging, ...restCallbacks } = useDragDropFiles(
+    handleFilesDropped,
+    fileDropDivRef
+  );
+
+  const headerElements = React.useMemo(
+    () => Object.values(headers).map((label) => <>{label}</>),
```

Comment:
memo wasn't needed.

<></> was needed because it could be localized string or jsx element. Removed empty fragment from here but too instead in source, only where string is used